tools/accuracy_checker/openvino/tools/accuracy_checker/evaluators/custom_evaluators/opennmt_encoder_decoder_generator_evaluator.py

```diff
@@ -0,0 +1,686 @@
+import pickle
+from functools import partial
+from collections import OrderedDict
+import numpy as np
+
+from .base_custom_evaluator import BaseCustomEvaluator
+from ...adapters import create_adapter
+from ...config import ConfigError
+from ...utils import contains_all, contains_any, read_pickle, get_path, extract_image_representations
+from ...logging import print_info
+
+
+class OpenNMTEvaluator(BaseCustomEvaluator):
+    def __init__(self, dataset_config, launcher, model, orig_config):
+        super().__init__(dataset_config, launcher, orig_config)
+        self.model = model
+        if hasattr(self.model, 'adapter'):
+            self.adapter_type = self.model.adapter.__provider__
+
+    @classmethod
+    def from_configs(cls, config, delayed_model_loading=False, orig_config=None):
+        dataset_config, launcher, _ = cls.get_dataset_and_launcher_info(config)
+        model = OpenNMTModel(
+            config.get('network_info', {}), launcher, config.get('_models', []), config.get('_model_is_blob'),
+            delayed_model_loading
+        )
+        return cls(dataset_config, launcher, model, orig_config)
+
+    def _process(self, output_callback, calculate_metrics, progress_reporter, metric_config, csv_file):
+        for batch_id, (batch_input_ids, batch_annotation, batch_inputs, batch_identifiers) in enumerate(self.dataset):
+            batch_inputs = self.preprocessor.process(batch_inputs, batch_annotation)
+            batch_inputs_extr, _ = extract_image_representations(batch_inputs)
+            encoder_callback = None
+            if output_callback:
+                encoder_callback = partial(output_callback, metrics_result=None, element_identifiers=batch_identifiers,
+                                           dataset_indices=batch_input_ids)
+            batch_raw_prediction, batch_prediction = self.model.predict(
+                batch_identifiers, batch_inputs_extr, encoder_callback=encoder_callback
+            )
+            metrics_result = self._get_metrics_result(batch_input_ids, batch_annotation, batch_prediction,
+                                                      calculate_metrics)
+            if output_callback:
+                output_callback(batch_raw_prediction[0], metrics_result=metrics_result,
+                                element_identifiers=batch_identifiers, dataset_indices=batch_input_ids)
+            self._update_progress(progress_reporter, metric_config, batch_id, len(batch_prediction), csv_file)
+
+
+class BaseModel:
+    default_model_suffix = ''
+
+    def __init__(self, network_info, launcher, delayed_model_loading=False):
+        self.network_info = network_info
+        self.launcher = launcher
+
+    def predict(self, identifiers, input_data):
+        raise NotImplementedError
+
+    def release(self):
+        pass
+
+    def fit_to_input(self, input_data):
+        pass
+
+    def propagate_output(self, output_data):
+        pass
+
+
+class BaseDLSDKModel(BaseModel):
```

Comment:
please use BaseDLSDKModel from base_models (at least inherit your class from it)

Reply:
Fixed